Formula/h/h5py.rb

```diff
@@ -0,0 +1,159 @@
+class H5py < Formula
+  desc "Read and write HDF5 files from Python"
+  homepage "https://www.h5py.org/"
+  url "https://files.pythonhosted.org/packages/37/fc/0b1825077a1c4c79a13984c59997e4b36702962df0bca420698f77b70b10/h5py-3.10.0.tar.gz"
+  sha256 "d93adc48ceeb33347eb24a634fb787efc7ae4644e6ea4ba733d099605045c049"
+  license "BSD-3-Clause"
+
+  depends_on "libcython" => :build
+  depends_on "pkg-config" => :build
+  depends_on "python-pkgconfig" => :build
+  depends_on "python-setuptools" => :build
+  depends_on "python@3.11" => [:build, :test]
+  depends_on "python@3.12" => [:build, :test]
+  depends_on "hdf5"
+  depends_on "numpy"
+
+  patch :DATA  # Patch is necessary to compile with cython >= 3.0.0
+  # Based on https://github.com/h5py/h5py/compare/master...takluyver:h5py:cython-3
```

Comment:
When can this patch be removed?

Reply:
When the issue https://github.com/h5py/h5py/issues/2268 is fixed upstream.

Reply:
That looks like it might not be any time soon given the timeline on that ticket. That makes me a bit reluctant to allow this into Homebrew/core, given our policy on patches.

Reply:
Well, sooner or later, they will have to do this. An alternative would be to add a formula `libcython@0.29` and use it to build `h5py`.

Reply:
Personally I'd say we wait with the formula instead.

> be maintained (i.e. the last release wasn’t ages ago, it works without patching on all Homebrew-supported OS versions [...]

https://docs.brew.sh/Acceptable-Formulae#niche-or-self-submitted-stuff